SerialPrograms/Source/PokemonSwSh/Inference/PokemonSwSh_BoxNatureDetector.cpp

```diff
@@ -23,49 +23,26 @@ namespace NintendoSwitch{
 namespace PokemonSwSh{
 
 BoxNatureDetector::BoxNatureDetector(VideoOverlay& overlay)
-    : m_box_atk(overlay, { 0.689, 0.198 + 1 * 0.0515, 0.84, 0.047})
-    , m_box_def(overlay, { 0.689, 0.198 + 2 * 0.0515, 0.84, 0.047})
-    , m_box_spatk(overlay, { 0.689, 0.198 + 3 * 0.0515, 0.84, 0.047})
-    , m_box_spdef(overlay, { 0.689, 0.198 + 4 * 0.0515, 0.84, 0.047})
-    , m_box_spd(overlay, { 0.689, 0.198 + 5 * 0.0515, 0.84, 0.047})
+    : m_box_atk(overlay, { 0.689, 0.198 + 1 * 0.0515, 0.084, 0.047})
+    , m_box_def(overlay, { 0.689, 0.198 + 2 * 0.0515, 0.084, 0.047})
+    , m_box_spatk(overlay, { 0.689, 0.198 + 3 * 0.0515, 0.084, 0.047})
+    , m_box_spdef(overlay, { 0.689, 0.198 + 4 * 0.0515, 0.084, 0.047})
+    , m_box_spd(overlay, { 0.689, 0.198 + 5 * 0.0515, 0.084, 0.047})
 {}
 
-NaturePlusMinus BoxNatureDetector::read(Logger& logger, const ImageViewRGB32& frame, const OverlayBoxScope& box){
+NaturePlusMinus BoxNatureDetector::read(Logger& logger, const ImageViewRGB32& frame, const ImageFloatBox& box){
+    const ImageStats region = image_stats(extract_box_reference(frame, box));
 
-    const auto region = extract_box_reference(frame, box);
-
-    // Retain only red pixels from region
-    const bool replace_color_within_range = false;
-    const ImageRGB32 red_region = filter_rgb32_range(
-        region,
-        combine_rgb(150, 0, 0), combine_rgb(255, 100, 100), Color(0), replace_color_within_range
-    );
-    const size_t num_red_pixels = image_stats(red_region).count;
-
-    // Retain only blue pixels from region
-    const ImageRGB32 blue_region = filter_rgb32_range(
-        region,
-        combine_rgb(0, 0, 180), combine_rgb(130, 130, 255), Color(0), replace_color_within_range
-    );
-    const size_t num_blue_pixels = image_stats(blue_region).count;
-
-    const double threshold = region.width() * region.height() * m_area_ratio_threshold;
-
-    if (PreloadSettings::debug().COLOR_CHECK) {
-        cout << "num_red_pixels: " << num_red_pixels << ", num_blue_pixels: " << num_blue_pixels
-            << ", region " << region.width() << " x " << region.height() << " threshold " << threshold << endl;
-
-        cout << "Save images to ./red_only.png and ./blue_only.png" << endl;
-        red_region.save("./red_only.png");
-        blue_region.save("./blue_only.png");
+    //cout << region.average.r << endl;
+    //cout << region.average.g << endl;
+    //cout << region.average.b << endl;
+    
+    if (region.average.b > region.average.r + 3) {
+        return NaturePlusMinus::MINUS;
     }
-
-    if (num_red_pixels > threshold) {
+    else if (region.average.r > region.average.b) {
```

Comment:
Is this actually reliable? There's only a 3 pixel window that can return neutral. It's easy for a capture cards to tint slightly in one direction.

What are the actual average RGB values you get for the different test images? When it's red, is the red much larger than the blue? Perhaps you can filter out the white background first, then average only the pixels that are left.